stream-chat-swift-docs/docs/guides/attachments-migration.md

```diff
@@ -0,0 +1,416 @@
+
+```swift
+// Get the first `.giphy` attachment
+if let giphy = message.giphyAttachments.first {
+  // Load and show gif right away
+  print(giphy.previewURL)
+}
+```
+
+</TabItem>
+</Tabs>
+
+### Link preview attachments
+
+The link attachment will be added to the message automatically if the message is sent with the text containing the URL.
+
+<Tabs
+  defaultValue="v4.x"
+  values={[
+    { label: 'v3.2', value: 'v3.2'},
+    { label: 'v4.x', value: 'v4.x'},
+  ]
+}>
+<TabItem value="v3.2">
+
+Giphy attachments are exposed as `ChatMessageDefaultAttachment`. Mandatory fields are **optinal** because of `ChatMessageDefaultAttachment` being used for all built-in attachment types.
+
+```swift
+// Get `.link(...)` attachments
+let linkAttachments = message.attachments
+  .compactMap { $0 as? ChatMessageDefaultAttachment }
+  .filter { $0.isLink }
+
+// Get the first attachment
+if let link = linkAttachments.first {
+  // Unwrap link URL
+  if let originalURL = link.ogURL {
+    // Show preview for originalURL
+    print(originalURL)
+  }
+}
+```
+
+</TabItem>
+<TabItem value="v4.x">
+
+Link preview attachments are exposed as `ChatMessageLinkAttachment`. Mandatory fields are **non-optinal** and can be accessed directly on attachment.
+
+```swift
+// Get the first `.linkPreview` attachment
+if let linkPreview = message.linkAttachments.first {
+  // Load and show preview
+  print(linkPreview.previewURL)
+}
+```
+
+</TabItem>
+</Tabs>
+
+### Custom attachments
+
+<Tabs
+  defaultValue="v4.x"
+  values={[
+    { label: 'v3.2', value: 'v3.2'},
+    { label: 'v4.x', value: 'v4.x'},
+  ]
+}>
+<TabItem value="v3.2">
+
+- all built-in attachments exposed as `ChatMessageRawAttachment`
+- custom payload is stored in `data: Data?` fields
+- custom data should be decoded manually `data`
+- attachment `id` is optional
+
+```swift
+// Custom attachment type has to be `Decodable`
+extension Product: Decodable { /* ... */ }
+
+// Get attachments of custom type
+let productAttachments = message.attachments
+  .compactMap { $0 as? ChatMessageRawAttachment }
+  .filter { $0.type == .custom("product") }
+
+// Get first custom attachment
+if let productAttachment = productAttachments.first {
+  // Unwrap attachment data
+  if let productData = productAttachment.data {
+    // Try to decode the custom type from data
+    let product = try JSONDecoder().decode(Product.self, from: productData)
+    // Handle custom attachment payload
+    print(product)
+  }
+}
+```
+
+</TabItem>
+<TabItem value="v4.x">
+
+- custom attachments are directly accessible on `ChatMessage`
+- custom payload fields are are directly on attachment thanks to `dynamicMemberLookup`
+- attachment `id` is not-optional 🎉
+
+```swift
+
+// It's recommended but not required to create a typealias for custom attachment type
+typealias ProductAttachment = _ChatMessageAttachment<Product>
+
+// Get attachments of custom type
+let productAttachments: [ProductAttachment] = message.attachments()
```

Comment:
IMO this is too magical notation :) Does the following work too?
```swift
let productAttachments = message.attachments(payloadType: ProductAttachment.self)
```

Reply:
Yep, it does 😄 ✅ Updated this line